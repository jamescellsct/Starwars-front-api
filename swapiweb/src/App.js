import React from "react";
import { Provider } from "react-redux";
import Routes from "./services/routes";
import store from "./store";

const App = () => (
  <Provider store={store}>
    <Routes />
  </Provider>
);

export default App;
