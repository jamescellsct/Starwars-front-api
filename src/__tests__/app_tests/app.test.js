import React from "react";

import App from "../../App";
import Routes from "../../services/routes";

describe("Testing App component", () => {
  it("Shold render correctly App component with Routes children", () => {
    const { children } = App().props;
    expect(children).toEqual(<Routes />);
    expect(children).not.toBe(null);
    expect(children).toMatchSnapshot();
  });

  it("Shold render correctly App component with initial state", () => {
    const { store } = App().props;
    const expectedState = {
      pagination: { page: 1 },
      peoples: null,
      planets: null,
      starships: null,
      vehicles: null
    };
    expect(store.getState()).toEqual(expectedState);
    expect(store.getState()).not.toBe(null);
    expect(store.getState()).toMatchSnapshot();
  });
});
