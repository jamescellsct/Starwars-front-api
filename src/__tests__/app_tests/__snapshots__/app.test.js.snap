// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component Shold render correctly App component with Routes children 1`] = `<Routes />`;

exports[`Testing App component Shold render correctly App component with initial state 1`] = `
Object {
  "pagination": Object {
    "page": 1,
  },
  "peoples": null,
  "planets": null,
  "starships": null,
  "vehicles": null,
}
`;
